sim/druid/guardian/TestGuardian.results

```diff
@@ -66,6 +66,60 @@ dps_results: {
   hps: 24676.64888
  }
 }
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-102292"
+ value: {
+  dps: 86731.27766
+  tps: 600970.21425
+  dtps: 59888.99742
+  hps: 21630.83821
+ }
+}
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-104476"
+ value: {
+  dps: 86858.2032
+  tps: 601812.64544
+  dtps: 59624.55446
+  hps: 21618.05186
+ }
+}
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-104725"
+ value: {
+  dps: 86593.1967
+  tps: 600044.27431
+  dtps: 59948.1643
+  hps: 21615.20745
+ }
+}
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-104974"
+ value: {
+  dps: 86450.9155
+  tps: 599082.01289
+  dtps: 59904.98145
+  hps: 21623.92641
+ }
+}
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-105223"
+ value: {
+  dps: 86787.71795
+  tps: 601344.70276
+  dtps: 59657.27598
+  hps: 21617.31514
+ }
+}
+dps_results: {
+ key: "TestGuardian-AllItems-AssuranceofConsequence-105472"
+ value: {
+  dps: 86923.60344
+  tps: 602218.85124
+  dtps: 59611.73071
+  hps: 21797.40387
+ }
+}
```

Comment:
You should modify the `RegisterAll()` method of `ItemVersionMap` so that only the highest ilvl variant is tested, just like we do within `NewProcStatBonusEffectWithVariants()`. In fact, if possible, we should unify the older `[]ItemVariant`  data structure with the newer `ItemVersionMap` structure, since the two are redundant with one another and are accomplishing the same thing.